src/OpenTelemetry/Trace/ExceptionProcessor.cs

```diff
@@ -23,19 +23,13 @@ public ExceptionProcessor()
 #else
         // When running on netstandard or similar the Marshal class is not a part of the netstandard API
         // but it would still most likely be available in the underlying framework, so use reflection to retrieve it.
-        try
-        {
-            var flags = BindingFlags.Static | BindingFlags.Public;
-            var method = typeof(Marshal).GetMethod("GetExceptionPointers", flags, null, Type.EmptyTypes, null)
-                ?? throw new InvalidOperationException("Marshal.GetExceptionPointers method could not be resolved reflectively.");
-            var lambda = Expression.Lambda<Func<IntPtr>>(Expression.Call(method));
-            this.fnGetExceptionPointers = lambda.Compile();
-        }
-        catch (Exception ex)
-        {
-            throw new NotSupportedException($"'{typeof(Marshal).FullName}.GetExceptionPointers' is not supported", ex);
-        }
+        var flags = BindingFlags.Static | BindingFlags.Public;
+        var method = typeof(Marshal).GetMethod("GetExceptionPointers", flags, null, Type.EmptyTypes, null)
+            ?? throw new InvalidOperationException("Marshal.GetExceptionPointers method could not be resolved reflectively.");
```

Comment:
Why would we want to throw InvalidOperationException here?
What does this message mean for the user (e.g. how would the user know what they are supposed to do)?

Reply:
The `InvalidOperationException` would be caught and used as the inner exception of the `NotSupportedException` thrown [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Trace/Builder/TracerProviderBuilderSdk.cs#L175). The `NotSupportedException` has the relevant message for the user.

Reply:
Thanks!